sdk/trace/batch_span_processor.go

```diff
@@ -0,0 +1,175 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trace
+
+import (
+	"context"
+	"errors"
+	"time"
+)
+
+const (
+	defaultMaxQueueSize         = 2048
+	defaultScheduledDelayMillis = time.Duration(5000 * time.Millisecond)
+	defaultMaxExportBatchSize   = 512
+)
+
+var (
+	errNilExporter = errors.New("exporter is nil")
+)
+
+type BatchSpanProcessorOption struct {
```

Comment:
nit: should this be `BatchSpanProcessorOptions` instead? 

As an aside, I saw there is an issue (#146) to make the Jaeger exporter take functional options instead of a struct options. I think if we are going with functional options, these options should also be functional.

Reply:
sure. Will convert to functional options.